### PR TITLE
Hardening

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: src/clients/inst_kickoff.rb [hunk 1]
```diff
@@ -44,8 +44,8 @@ def main
           # When upgrading system, remove devs.rpm if installed
```
Comment (lslezak) [inline src/clients/inst_kickoff.rb]: Hm, this has been added 12 years ago, the `devs` package has been dropped long time ago. https://maintenance.suse.de/maintained/?package=devs displays SLE10.

So I'd rather completely remove this obsolete code.

File: src/clients/inst_kickoff.rb [hunk 2]
```diff
@@ -474,17 +429,9 @@ def createmdadm
...
-            Ops.add(
-              Ops.add("chroot '", String.Quote(Installation.destdir)),
-              "' "
-            ),
-            "mdadm -Ds"
-          )
-        )
+      out = SCR.Execute(
+        path(".target.bash_output"),
+        "/usr/bin/chroot #{Installation.destdir.shellescape} mdadm -Ds"
```
Comment (lslezak) [inline src/clients/inst_kickoff.rb]: Do we still need this? Maybe it's already handled by the `libstorage-ng`... Again, this code is ~13 years old... :open_mouth:

File: src/modules/Packages.rb [hunk 3]
```diff
@@ -1227,7 +1228,7 @@ def ComputeSystemPackageList
     # Import GPG keys found in the inst-sys
     def ImportGPGKeys
       out = Convert.to_map(
-        SCR.Execute(path(".target.bash_output"), "/bin/ls -d /*.gpg")
+        SCR.Execute(path(".target.bash_output"), "/usr/bin/ls -d /*.gpg")
```
Comment (lslezak) [inline src/modules/Packages.rb]: NP: maybe we could use some native Ruby here, as it reads from inst-sys we do not need any chroot magic.

File: src/modules/Packages.rb [hunk 4]
```diff
@@ -1702,11 +1683,7 @@ def FindAndRememberAddOnProductsFiles(initial_repository)
           )
           WFM.Execute(
             path(".local.bash"),
-            Builtins.sformat(
-              "cp '%1' '%2'",
-              String.Quote(tmp_add_on_products),
-              String.Quote(filename)
-            )
+            "/usr/bin/cp #{tmp_add_on_products.shellescape} #{filename.shellescape}"
```
Comment (lslezak) [inline src/modules/Packages.rb]: NP: Use [::FileUtils.cp](https://ruby-doc.org/stdlib-2.5.1/libdoc/fileutils/rdoc/FileUtils.html#method-c-cp) here?

File: src/modules/Packages.rb [hunk 5]
```diff
@@ -1367,18 +1360,12 @@ def FindAndCopySlideDir(our_slidedir, source, search_for_dir, lang_long,
...
 
-        copy_command = Builtins.sformat(
-          "cp -r '%1/%2/pic'/* '%3'",
-          String.Quote(imagesdir),
-          String.Quote(search_for_dir),
-          String.Quote(our_imagesdir)
-        )
+        source_path = File.join(imagesdir, search_for_dir, "pic", "*")
+
+        copy_command = "/usr/bin/cp -r #{source_path.shellescape} #{our_imagesdir.shellescape}"
```
Comment (lslezak) [inline src/modules/Packages.rb]: NP: Use [::FileUtils.cp_r](https://ruby-doc.org/stdlib-2.5.1/libdoc/fileutils/rdoc/FileUtils.html#method-c-cp_r) here?

File: src/modules/SourceDialogs.rb [hunk 6]
```diff
@@ -1966,11 +1966,10 @@ def ServerWidget
     def IsAnyNetworkAvailable
```
Comment (lslezak) [inline src/modules/SourceDialogs.rb]: We should replace the implementation by just calling [NetworkService.isNetworkRunning](https://github.com/yast/yast-yast2/blob/master/library/network/src/modules/NetworkService.rb#L280) or even drop it and update the callers to use it instead...

File: src/modules/SourceManager.rb [hunk 7]
```diff
@@ -751,14 +753,9 @@ def InstInitSourceMoveDownloadArea
...
-        SCR.Execute(
-          path(".target.bash"),
-          Builtins.sformat(
-            "test -d '%1' || mkdir -p '%1'",
-            String.Quote(download_dir)
-          )
-        )
+      successful = SCR.Execute(
+        path(".target.bash"),
+        "test -d #{download_dir.shellescape} || /usr/bin/mkdir -p #{download_dir.shellescape}"
```
Comment (lslezak) [inline src/modules/SourceManager.rb]: I guess we could drop that `test` as `mkdir -p` does not fail if it already exists.

File: src/modules/Packages.rb (hunk 3 above)
Comment (jreidinger) [reply src/modules/Packages.rb]: well, I do not where it is trivial enough, as I do not want to break it too much.

File: src/clients/inst_kickoff.rb (hunk 2 above)
Comment (jreidinger) [reply src/clients/inst_kickoff.rb]: I do not know. @ancorgs ^^^

File: src/clients/inst_kickoff.rb (hunk 1 above)
Comment (jreidinger) [reply src/clients/inst_kickoff.rb]: ok, I do not know it.

File: src/modules/Packages.rb (hunk 5 above)
Comment (jreidinger) [reply src/modules/Packages.rb]: if we are sure that it cannot fail, then it is ok for me to change it.

File: src/modules/SourceDialogs.rb (hunk 6 above)
Comment (jreidinger) [reply src/modules/SourceDialogs.rb]: ok

File: src/modules/SourceManager.rb (hunk 7 above)
Comment (jreidinger) [reply src/modules/SourceManager.rb]: fine for me.

File: src/modules/Packages.rb (hunk 5 above)
Comment (jreidinger) [reply src/modules/Packages.rb]: well, this won't work with fileutils as it uses "*" expansion in path. So I keep it as it is.

File: src/clients/inst_kickoff.rb [hunk 8]
```diff
@@ -34,21 +34,7 @@ def main
...
-        else
-          # When upgrading system, remove devs.rpm if installed
-          LocalCommand(
-            Builtins.sformat(
-              "/bin/rpm --root %1 -q 'devs' && /bin/rpm --nodeps --root %1 -e 'devs'",
-              Installation.destdir.shellescape
-            )
-          )
-
+        if Installation.destdir != "/"
```
Comment (lslezak) [inline src/clients/inst_kickoff.rb]: NP: Merge these two conditions? `if Mode.update && Installation.destdir != "/"`

Comment (jreidinger) [reply src/clients/inst_kickoff.rb]: I am still not sure if we do not have more robust way to testing online upgrade. But fine for me to merge it.

**Pull request merged**